### DISCUSSION TITLE
How to add new columns

### DISCUSSION THREAD

**Discussion by OjoFemiJoseph** [delta-io/delta-rs#2333] [Q&A]
how do i add new column to existing delta tables

self.dt = DeltaTable('data/deltaname')

                (
                    self.dt.merge(
                        source=self.pandas_df,
                        predicate="s._id = t._id",
                        source_alias="s",
                        target_alias="t",
                    )
                    .when_matched_update_all()
                    .when_not_matched_insert_all()
                    .execute()
                )

i noticed that if my dataframe has new columns, it doesnt add it. what do i need to add for new columns to be added ?

**Comment by ion-elgreco** [comment]
Current only appends or overwrites  can be done to add new columns, schema evolution as it's called is not possible yet in MERGE
